WordPress/gutenberg
# ChildLayoutControl: Use units defined in theme.json

## What?

Fixes #67750

## Why?

Currently, this component does not pass a `unit` prop to its internal `UnitControl`, so [the default units](https://github.com/WordPress/gutenberg/blob/7649b9927f8d9965d4dd498e9d7f7bf5bb82a67d/packages/components/src/unit-control/index.tsx#L55) will always be used. Like most other controls, I think it should respect the unit values ​​defined in theme.json.

## Testing Instructions

- Activate the theme with the following theme.json defined:
  ```json
  {
  	"version": 3,
  	"settings": {
  		"appearanceTools": true,
  		"layout": {
  			"contentSize": "840px",
  			"wideSize": "1100px"
  		},
  		"spacing": {
  			"units": [ "px", "em", "rem", "vh", "vw", "%", "dvh", "dvw" ]
  		}
  	}
  }
- Insert any block inside the Row block.
- Access the Dimensions panel.
- Select "Fixed".
- You should be able to select the units defined in theme.json.

## Screenshots or screencast 

![after](https://github.com/user-attachments/assets/df405dfa-0091-47c8-82f1-248e0843f276)
